webapp/tests/karma/ts/services/rdtoolkit.service.spec.ts

```diff
@@ -0,0 +1,153 @@
+      expect(medicMobileAndroid.rdToolkit_provisionRDTest.args[0]).to.have.members([
+        'a1',
+        'b1',
+        'Anne',
+        'mal_pf',
+        'https//server'
+      ]);
+      expect(result instanceof Promise).to.equal(true);
+      expect(consoleErrorMock.callCount).to.equal(0);
+    });
+
+    it('should catch exception when provisioning test', () => {
+      medicMobileAndroid.rdToolkit_provisionRDTest.throws(new Error('some error'));
+      service.provisionRDTest(
+        'a1',
+        'b1',
+        'Anne',
+        'mal_pf',
+        'https//server'
+      );
+
+      expect(medicMobileAndroid.rdToolkit_provisionRDTest.callCount).to.equal(1);
+      expect(medicMobileAndroid.rdToolkit_provisionRDTest.args[0]).to.have.members([
+        'a1',
+        'b1',
+        'Anne',
+        'mal_pf',
+        'https//server'
+      ]);
+      expect(consoleErrorMock.callCount).to.equal(1);
+      expect(consoleErrorMock.args[0][0]).to.equal(
+        'Error when provisioning RD Test: ',
+        'Error: some error'
+      );
+    });
+  });
+
+  describe('captureRDTest()', () => {
+    it('should return promise and capture test result', () => {
+      const result = service.captureRDTest('a1');
+
+      expect(medicMobileAndroid.rdToolkit_captureRDTest.callCount).to.equal(1);
+      expect(medicMobileAndroid.rdToolkit_captureRDTest.args[0]).to.have.members(['a1']);
+      expect(result instanceof Promise).to.equal(true);
+      expect(consoleErrorMock.callCount).to.equal(0);
+    });
+
+    it('should catch exception when capturing test result', () => {
+      medicMobileAndroid.rdToolkit_captureRDTest.throws(new Error('some error'));
+      service.captureRDTest('a1');
+
+      expect(medicMobileAndroid.rdToolkit_captureRDTest.callCount).to.equal(1);
+      expect(medicMobileAndroid.rdToolkit_captureRDTest.args[0]).to.have.members(['a1']);
+      expect(consoleErrorMock.callCount).to.equal(1);
+      expect(consoleErrorMock.args[0][0]).to.equal(
+        'Error when capturing RD Test: ',
+        'Error: some error'
+      );
+    });
+  });
+
+  it('should resolve the response after provisioning test', () => {
+    service.provisionRDTest(
+      'a1',
+      'b1',
+      'Anne',
+      'mal_pf',
+      'https//server'
+    );
+    const provisionTestResolve = sinon.spy(service, 'provisionTestResolve');
+
+    service.resolveProvisionedTest({ sessionId: 'a1' });
+
+    expect(provisionTestResolve.callCount).to.equal(1);
+    expect(provisionTestResolve.args[0][0]).to.deep.equal({ sessionId: 'a1' });
+  });
+
+  it('should resolve the response after capturing response test', () => {
+    service.captureRDTest('a1');
+    const captureTestResolve = sinon.spy(service, 'captureTestResolve');
+
+    service.resolveCapturedTest({ sessionId: 'a1' });
+
+    expect(captureTestResolve.callCount).to.equal(1);
+    expect(captureTestResolve.args[0][0]).to.deep.equal({ sessionId: 'a1' });
+  });
```

Comment:
I think we should have tests that handle both actions: of requesting the promise and resolving it (for both cases). Is that possible?

Reply:
Done!